commit 407e7c90a97143d436ad3a6af6bbfa431c849191
Author: Marc Dionne <marc.c.dionne@gmail.com>
Date:   Sat Jun 2 21:35:53 2012 -0400

    Linux 3.5: encode_fh API change
    
    The encode_fh export operation now expects two inode arguments
    instead of a dentry and a "connectable" flag.  Use the inode of
    the dentry we're interested in, and NULL as the parent inode which
    is the same as passing a 0 flag in the previous API.
    
    Change-Id: I05cf146fb2a4bacdca20a9f108d04ccb11530804
    Reviewed-on: http://gerrit.openafs.org/7523
    Tested-by: BuildBot <buildbot@rampaginggeek.com>
    Reviewed-by: Derrick Brashear <shadow@dementix.org>
    (cherry picked from commit 5227148ae17949705487ea673d558ebfe143e635)
    Reviewed-on: http://gerrit.openafs.org/7579

--- a/src/afs/LINUX/osi_compat.h
+++ b/src/afs/LINUX/osi_compat.h
@@ -333,7 +333,7 @@ afs_get_dentry_from_fh(struct super_block *afs_cacheSBp, afs_dcache_id_t *ainode
 static inline int
 afs_get_fh_from_dentry(struct dentry *dp, afs_ufs_dcache_id_t *ainode, int *max_lenp) {
     if (dp->d_sb->s_export_op->encode_fh)
+        return dp->d_sb->s_export_op->encode_fh(dp->d_inode, &ainode->raw[0], max_lenp, NULL);
-        return dp->d_sb->s_export_op->encode_fh(dp, &ainode->raw[0], max_lenp, 0);
 #if defined(NEW_EXPORT_OPS)
     /* If fs doesn't provide an encode_fh method, assume the default INO32 type */
     *max_lenp = sizeof(struct fid)/4;
